### ISSUE TITLE
Test failure with race flag on macOS Monterey 12.1 (21C52) - SIGABRT Attempt to use unknown class

### ISSUE THREAD

**Issue by piersy** [ethereum/go-ethereum#24225]
#### System information

Geth version: `geth version`
```
Geth
Version: 1.10.16-unstable
Git Commit: c006261758727736337bfc4835a606cdc738ac90
Git Commit Date: 20220111
Architecture: amd64
Go Version: go1.16.12
Operating System: darwin
GOPATH=/Users/piersy/go
GOROOT=go
```
OS & Version: macOS Monterey 12.1 (21C52)

#### Expected behaviour

Tests should pass

#### Actual behaviour

Many tests are failing as evidenced by the backtrace below (restricted to a single package for brevity), although there appear to be 29 SIGABRT failures in total all arising from use of the notify package.

#### Steps to reproduce the behaviour

`go test -race ./...`

#### Backtrace

```
$ go test -race ./accounts/abi/bind
objc[49518]: Attempt to use unknown class 0x7ff84243d260.
SIGABRT: abort
PC=0x7ff800bf5dfe m=0 sigcode=0
signal arrived during cgo execution

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x48c0950, 0xc000139920, 0x49adc80)
	/Users/piersy/sdk/go1.16.12/src/runtime/cgocall.go:154 +0x7a fp=0xc0001398f0 sp=0xc0001398b8 pc=0x40083ba
github.com/rjeczalik/notify._Cfunc_FSEventsGetCurrentEventId(0x0)
	_cgo_gotypes.go:378 +0x68 fp=0xc000139920 sp=0xc0001398f0 pc=0x45831a8
github.com/rjeczalik/notify.init()
	/Users/piersy/go/pkg/mod/github.com/rjeczalik/notify@v0.9.1/watcher_fsevents_cgo.go:41 +0xc57 fp=0xc000139a48 sp=0xc000139920 pc=0x4588177
runtime.doInit(0x4ea9da0)
	/Users/piersy/sdk/go1.16.12/src/runtime/proc.go:6315 +0xec fp=0xc000139b98 sp=0xc000139a48 pc=0x404e28c
runtime.doInit(0x4eae140)
	/Users/piersy/sdk/go1.16.12/src/runtime/proc.go:6292 +0x72 fp=0xc000139ce8 sp=0xc000139b98 pc=0x404e212
runtime.doInit(0x4eadb40)
	/Users/piersy/sdk/go1.16.12/src/runtime/proc.go:6292 +0x72 fp=0xc000139e38 sp=0xc000139ce8 pc=0x404e212
runtime.doInit(0x4ea6820)
	/Users/piersy/sdk/go1.16.12/src/runtime/proc.go:6292 +0x72 fp=0xc000139f88 sp=0xc000139e38 pc=0x404e212
runtime.main()
	/Users/piersy/sdk/go1.16.12/src/runtime/proc.go:208 +0x205 fp=0xc000139fe0 sp=0xc000139f88 pc=0x4040145
runtime.goexit()
	/Users/piersy/sdk/go1.16.12/src/runtime/asm_amd64.s:1371 +0x1 fp=0xc000139fe8 sp=0xc000139fe0 pc=0x4079481

goroutine 21 [chan receive]:
github.com/rjeczalik/notify.(*recursiveTree).dispatch(0xc00015a280)
	/Users/piersy/go/pkg/mod/github.com/rjeczalik/notify@v0.9.1/tree_recursive.go:125 +0x105
created by github.com/rjeczalik/notify.newRecursiveTree
	/Users/piersy/go/pkg/mod/github.com/rjeczalik/notify@v0.9.1/tree_recursive.go:119 +0x21c

rax    0x2000209
rbx    0x80
rcx    0x7ff7bfefe6a8
rdx    0x0
rdi    0x8
rsi    0x1
rbp    0x7ff7bfefe6f0
rsp    0x7ff7bfefe6a8
r8     0x600002c00000
r9     0x80
r10    0x0
r11    0x246
r12    0x0
r13    0x0
r14    0x1
r15    0x8
rip    0x7ff800bf5dfe
rflags 0x246
cs     0x7
fs     0x0
gs     0x0
FAIL	github.com/ethereum/go-ethereum/accounts/abi/bind	0.569s
FAIL
```

When submitting logs: please submit them as text and not screenshots.

**Comment by fjl** [comment]
Thanks. I think this is a problem with https://github.com/rjeczalik/notify

**Comment by piersy** [comment]
@fjl Yep https://github.com/rjeczalik/notify seems to be the source of the problem

**Comment by fjl** [comment]
It's this issue: https://github.com/golang/go/issues/49138 

The fix has been backported into Go 1.17.6.
